lib/src/main/java/com/dynatrace/metric/util/DynatraceMetadataEnricher.java

```diff
@@ -0,0 +1,192 @@
+   * @return The string containing the filename or null if the file is empty.
+   * @throws IOException if an error occurs during reading of the file.
+   */
+  static String getMetadataFileName(Reader fileContents) throws IOException {
+    if (fileContents == null) {
+      throw new IOException("passed Reader cannot be null.");
+    }
+
+    String oneAgentMetadataFileName = null;
+
+    try (BufferedReader reader = new BufferedReader(fileContents)) {
+      String line = reader.readLine();
+      if (line != null && !line.isEmpty()) {
+        oneAgentMetadataFileName = line;
+      }
+    }
+    return oneAgentMetadataFileName;
+  }
+
+  /**
+   * Read the actual content of the metadata file.
+   *
+   * @param fileContents A {@link Reader} object containing the metadata contents.
+   * @return A {@link List<String>} containing the {@link String#trim() trimmed} lines.
+   * @throws IOException if an error occurs during reading of the file.
+   */
+  static List<String> getDynatraceMetadataFileContents(Reader fileContents) throws IOException {
+    if (fileContents == null) {
+      throw new IOException("passed Reader cannot be null.");
+    }
+    try (BufferedReader reader = new BufferedReader(fileContents)) {
+      return reader.lines().map(String::trim).collect(Collectors.toList());
+    }
+  }
+
+  /**
+   * A helper function that returns whether file exists.
+   *
+   * @param alternativeMetadataFileName The file name.
+   * @return true if the file exists and is readable and false otherwise.
+   */
+  static boolean fileExistsAndIsReadable(String alternativeMetadataFileName) {
+    try {
+      File file = new File(alternativeMetadataFileName);
+      if (file.exists() && Files.isReadable(file.toPath())) {
+        return true;
+      }
+    } catch (Throwable ignored) {
+      // something went wrong, but at this point we don't care what it was.
+    }
+    return false;
+  }
+
+  /**
+   * Gets the file location of the metadata file from the indirection file and reads the contents
+   * thereof. If the indirection file does not exist, falls back to the alternative metadata file.
+   *
+   * @return A {@link List<String>} representing the contents of the Dynatrace metadata file.
+   *     Leading and trailing whitespaces are {@link String#trim() trimmed} for each of the lines.
+   */
+  static List<String> getMetadataFileContentWithRedirection(
+      String indirectionFileName, String alternativeMetadataFilename) {
+    String metadataFileName = null;
+
+    try (Reader indirectionFileReader = new FileReader(indirectionFileName)) {
+      metadataFileName = getMetadataFileName(indirectionFileReader);
+    } catch (FileNotFoundException e) {
+      logger.info("Indirection file not found. This is normal if OneAgent is not installed.");
+    } catch (IOException e) {
+      logger.info(
+          String.format(
+              "Error while trying to read contents of OneAgent indirection file: %s",
+              e.getMessage()));
```

Comment:
```suggestion
              e);
```
With `getMessage` you only get the message (which often is empty) but lose the class name. The `toString` representation is usually more helpful.